libs/providers/flipt-web/src/lib/flipt-web-provider.ts

```diff
@@ -0,0 +1,263 @@
+    defaultValue: boolean,
+    context: EvaluationContext,
+  ): ResolutionDetails<boolean> {
+    const mergedContext = { ...this._globalContext, ...context };
+    const evalContext: Record<string, string> = transformContext(mergedContext);
+
+    try {
+      const resp = this._client?.evaluateBoolean(flagKey, mergedContext.targetingKey ?? '', evalContext);
+
+      if (resp?.status === 'failure') {
+        return {
+          value: defaultValue,
+          errorCode: ErrorCode.GENERAL,
+          errorMessage: resp.error_message,
+          reason: StandardResolutionReasons.ERROR,
+        };
+      }
+
+      const result = resp?.result;
+
+      switch (result?.reason) {
+        case EvaluationReason.DEFAULT:
+          return {
+            value: result?.enabled,
+            reason: StandardResolutionReasons.TARGETING_MATCH,
+          };
+        case EvaluationReason.MATCH:
+          return {
+            value: result?.enabled,
+            reason: StandardResolutionReasons.TARGETING_MATCH,
+          };
+        case EvaluationReason.UNKNOWN:
+          return {
+            value: defaultValue,
+            reason: StandardResolutionReasons.UNKNOWN,
+          };
+        default:
+          return {
+            value: defaultValue,
+            reason: StandardResolutionReasons.DEFAULT,
+          };
+      }
+    } catch (e) {
+      throw new GeneralError(getErrorMessage(e));
+    }
+  }
+
+  resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    context: EvaluationContext,
+  ): ResolutionDetails<string> {
+    const mergedContext = { ...this._globalContext, ...context };
+    const value = this.resolveFlagHelper(flagKey, 'string', defaultValue, mergedContext);
+    return value as ResolutionDetails<string>;
+  }
+
+  resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+    context: EvaluationContext,
+  ): ResolutionDetails<number> {
+    const mergedContext = { ...this._globalContext, ...context };
+    const value = this.resolveFlagHelper(flagKey, 'number', defaultValue, mergedContext);
+    return value as ResolutionDetails<number>;
+  }
+
+  resolveObjectEvaluation<U extends JsonValue>(
+    flagKey: string,
+    defaultValue: U,
+    context: EvaluationContext,
+  ): ResolutionDetails<U> {
+    const mergedContext = { ...this._globalContext, ...context };
+    const value = this.resolveFlagHelper(flagKey, 'json', defaultValue, mergedContext);
+    return value as ResolutionDetails<U>;
+  }
+
+  private resolveFlagHelper<U extends JsonValue>(
+    flagKey: string,
+    flagType: PrimitiveTypeName,
+    defaultValue: PrimitiveType | U,
+    context: EvaluationContext,
+  ): ResolutionDetails<PrimitiveType | U> {
+    const evalContext: Record<string, string> = transformContext(context);
+
+    try {
+      const resp = this._client?.evaluateVariant(flagKey, context.targetingKey ?? '', evalContext);
+
+      if (resp?.status === 'failure') {
+        return {
```

Comment:
You can throw the general error with a message. The SDK will handle the response for you.